test: Fix test cache directory conflict

Milly opened the pull request:
Previously, some tests failed when run in parallel or with shuffling due to conflicts in the test cache directory.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Updated test cache directory path to improve test isolation; no changes to test logic or assertions.
* **Chores**
  * Enhanced coverage reporting by broadening exclusion patterns for test-related files.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Milly committed:
test: Fix test cache directory conflict

Previously, some tests failed when run in parallel or with shuffling due
to conflicts in the test cache directory.

lambdalisue commented on import_map_importer_import_meta_url_test.ts:
```diff
@@ -8,7 +8,7 @@ import type { ImportMap } from "./import_map.ts";
   try {
     const cacheDir = fromFileUrl(
-      new URL("./.test_cache", import.meta.url),
+      new URL("./.test_cache_meta", import.meta.url),
```
We need to ignore `./test_cache_meta` in `deno.jsonc` like `./test_cache`.

https://github.com/lambdalisue/deno-import-map-importer/blob/1cc27e732c357b3f124cd38d31d7ad9955f93255/deno.jsonc#L31

Milly replied:
Fixed. Note that the `--exclude` option requires a backslash for dot matching as it uses regex patterns.

Milly committed:
fix: Update coverage exclusion pattern

Use regex pattern '\.test.*' to exclude both .test_cache and .test_cache_meta directories
from coverage reports, as both are now used for testing purposes
